### ISSUE TITLE
circles drawn in grid viewports are too big

### ISSUE THREAD

**Issue by trevorld** [r-lib/svglite#93]
I'm observing ``svglite`` drawing circles that are too big when drawing circles in grid viewports.  This issue doesn't seem to affect rectangles.  I don't observe this behaviour with the cairo family of graphic devices.

Here is a minimal example where the circles should touch (but not significantly overlap):  

```
library("grid")

draw_two_circles <- function() {
    pushViewport(viewport(width=unit(2, "inches"), height=unit(1, "inches")))
    pushViewport(viewport(x=0.25, width=unit(1, "inches"), height=unit(1, "inches")))
    grid.circle()
    popViewport()
    pushViewport(viewport(x=0.75, width=unit(1, "inches"), height=unit(1, "inches")))
    grid.circle()
    popViewport()
    popViewport()
}

svg("svg_test.svg"); draw_two_circles(); dev.off()
svglite::svglite("svglite_test.svg"); draw_two_circles(); dev.off()
```

![svglite_test](https://user-images.githubusercontent.com/2433259/48865505-a97fef80-ed84-11e8-915a-a80acfb6a439.png)

**Comment by trevorld** [comment]
Here is perhaps a better minimal example where the circle should be completely hidden by a rectangle (filling up the entire viewport) but instead when drawn in ``svglite`` the circle is too big and parts of it become visible (escaping the boundaries of its viewport).

```
library("grid")
draw_hidden_circle <- function() {
    pushViewport(viewport(width=0.5, height=0.5))
    grid.circle(gp=gpar(fill="blue"))
    grid.rect(gp=gpar(fill="yellow"))
    popViewport()
}


svg("svg_test.svg"); draw_hidden_circle(); dev.off()
svglite::svglite("svglite_test.svg"); draw_hidden_circle(); dev.off()
```

![svglite_test](https://user-images.githubusercontent.com/2433259/48860354-8a2d9600-ed75-11e8-9f2f-1f873fe3c0f4.png)

**Comment by vandenman** [comment]
I'm also quite affected by this issue hence I'm reviving the thread (just tell me if that's a bad idea!).

Using some trial and error, I was able to find the source of the issue:

I modified the example above so that the resulting svg is square (`svglite::svglite("~/svglite_test.svg", height = 7, width = 7)`). Examining the resulting svg, the key issue is this:
```svg
                                   ↓  pt???
<circle cx='252.00' cy='252.00' r='126.00pt' style='stroke-width: 0.75; fill: #0000FF;' />
<rect x='126.00' y='126.00' width='252.00' height='252.00' style='stroke-width: 0.75; fill: #FFFF00;' />
```
The radius is rendered using pt. However, I think this is unnecessary. If I remove the pt and examine the result in Inkscape then the circle appears to have the correct size and is completely covered by the square.

I'm happy to create a fork in an attempt to fix this. However, this change may very well break things downstream (or at least make it look differently, causing unit tests to fail). Any pointers on how to proceed?
